### ISSUE TITLE
Query result does not contain columns in expected order

### ISSUE THREAD

**Issue by hleskien** [Hi-Fi/robotframework-dblibrary#19]
Hi,

if I do a SQL query with columns in a certain order, the database delivers them correctly, but the order is lost (random) in the result. The reason is using HashMap instead of LinkedHashMap which would keep the insertion order. I don't know yet whether this works together with Robot Framework.

Example:

``
SELECT COLUMNA, COLUMNB, COLUMNC FROM TABLEABC
``
->
``
@{result} = [
	{ COLUMNB: 2, COLUMNA: 1, COLUMNC: 3 },
	{ COLUMNB: 5, COLUMNA: 4, COLUMNC: 6 },
	{ COLUMNB: 8, COLUMNA: 7, COLUMNC: 9 }
]
``

Kind regards,

Holger

**Comment by Hi-Fi** [comment]
At least when I made test with H2 I get order correctly.

Test case:
```
Result Order Check with Execute SQL
    [Documentation]    Checks that query execution returns results in correct order
    ${data}    Execute SQL    Select ID, NAME from DemoTable    
    Log    ${data}
    ${data}    Execute SQL    Select NAME, ID from DemoTable    
    Log    ${data}
```

Logged output (in order):
```
[{ID=1, NAME=Donny Darko}, {ID=2, NAME=Darth Vader}]
[{NAME=Donny Darko, ID=1}, {NAME=Darth Vader, ID=2}]
```

Order seems to be keeping same also when storing results as array:
```
    @{data}    Execute SQL    Select ID, NAME from DemoTable    
    @{data}    Execute SQL    Select NAME, ID from DemoTable 
```

If you have the issue with ordering, can you provide more information about the DB, and preferably create test case that shows the error with H2 (if reproducible with that).

**Comment by hleskien** [comment]
I made a similar test with the DB Library source code:
```
@Test
public void checkExecuteSQLColumnOrder() throws Exception {
	List<HashMap<String, Object>> result2 = query.executeSql("SELECT EMail, Id, LastPosting, Name, Postings, State FROM MySampleTable");
	printResult(result2);
	List<HashMap<String, Object>> result3 = query.executeSql("SELECT State, Postings, Name, LastPosting, Id, EMail FROM MySampleTable");
	printResult(result3);
}

private void printResult(List<HashMap<String, Object>> result){
	for (HashMap row : result){
		System.out.println(row.toString());
	}
}
```
->
```
{STATE=1, ID=1, LASTPOSTING=2010-02-26 12:42:58.0, EMAIL=donny.darko@robot.org, NAME=Donny Darko, POSTINGS=1001}
{STATE=2, ID=2, LASTPOSTING=2010-02-27 11:41:45.0, EMAIL=darth.vader@starwars.universe, NAME=Darth Vader, POSTINGS=123}
{STATE=1, LASTPOSTING=2010-02-26 12:42:58.0, ID=1, EMAIL=donny.darko@robot.org, POSTINGS=1001, NAME=Donny Darko}
{STATE=2, LASTPOSTING=2010-02-27 11:41:45.0, ID=2, EMAIL=darth.vader@starwars.universe, POSTINGS=123, NAME=Darth Vader}
```
So I get a different result with the same database. When I exchange HashMap with LinkedHashMap in the test and all called methods, it works correctly.

**Comment by hleskien** [comment]
Thank you! I'll test it with our production code and give feedback.

**Comment by Hi-Fi** [comment]
Released version containing this as 3.1.3.

**Comment by hleskien** [comment]
I tested the development version: it works as intended. Thank you very much!
